src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -73,17 +80,19 @@
     /**
      * Constructor.
      *
-     * @param string $environment The environment
-     * @param bool   $debug       Whether to enable debugging or not
+     * @param string                      $environment        The environment
+     * @param bool                        $debug              Whether to enable debugging or not
+     * @param ConfigCacheFactoryInterface $configCacheFactory The factory used to create the config cache
      *
      * @api
      */
-    public function __construct($environment, $debug)
+    public function __construct($environment, $debug, ConfigCacheFactoryInterface $configCacheFactory = null)
```

Comment:
Not sure if we will accept this PR as I don't see the use case but in any case, this should be moved to a method instead of a constructor argument. I would even create a protected method like the getCacheDir() method, so configuration happen in the AppKernel class instead of whenever you need to create a Kernel instance (think tests for instance).